### PR TITLE
Otbn rtl

### PR DESCRIPTION
**Pull request by GregAC**
This gives us some initial working RTL for OTBN along with a standalone simulation using verilator which can run a reasonable subset of the base rv32i instruction set (everything bar the memory and control instructions I think) it comes from the otbn-dev branch.

It has been reviewed going into otbn-dev already so should just need a light touch here.

There will be a follow up PR with a smoke test that runs in CI using the standalone simulator.

### CODE REVIEW

File: hw/ip/otbn/README.md [hunk 1]
```diff
@@ -13,11 +13,37 @@ through the [GitHub issue tracker](https://github.com/lowRISC/opentitan/issues).
...
 ## Develop OTBN
 
-### Build the RTL implementation
+### Run the standalone simulation
 
-To build the RTL implementation of OTBN in a simulation, run `fusesoc` without
-passing the `OTBN_MODEL` flag. For example, the Verilator simulation can be
-compiled as follows.
+*Note that OTBN is still in the early development stages so this simulation will
+do very little currently*
```
Comment (rswarbrick) [inline hw/ip/otbn/README.md:19]: Nit: This should be "will currently do very little".

File: hw/ip/otbn/README.md [hunk 2]
```diff
@@ -13,11 +13,37 @@ through the [GitHub issue tracker](https://github.com/lowRISC/opentitan/issues).
...
+
+A standalone environment to run OTBN alone in verilator is included. It is built
+with `fusesoc`:
```
Comment (rswarbrick) [inline hw/ip/otbn/README.md:22]: Nit: The "is" in the two sentences mean different things, which looks a bit odd. Probably easiest to change the second sentence to an instruction: "Build it with `fusesoc` as follows:"

File: hw/ip/otbn/README.md [hunk 3]
```diff
@@ -13,11 +13,37 @@ through the [GitHub issue tracker](https://github.com/lowRISC/opentitan/issues).
...
+
+```sh
+fusesoc --cores-root=. run --target=sim --setup --build lowrisc:ip:otbn_top_sim
+```
+
+It includes functionality to set the initial DMem and IMem contents with a .vmem
+or .elf file. The start address is hard coded to 0. Modify the `ImemStartAddr`
```
Comment (rswarbrick) [inline hw/ip/otbn/README.md:29]: Probably drop the reference to ELF files? A `.elf` suffix is non-standard, and we don't support a single ELF "out of the box" anyway at the moment. And the example below uses `.vmem` files anyway.

**Comment by GregAC** [comment]
Thanks for the review @rswarbrick 

I've dropped the "[otbn] Use generic insn_op for all ALU operations" commit entirely as `insn_op` was removed entirely in a later commit.

**Comment by sriyerg** [comment]
Would you please add otbn to the lint targets? You will need to add an entry for otbn to this file:
`hw/top_earlgrey/lint/top_earlgrey_lint_cfgs.hjson` 

The style lint runs as a part of CI.

**Comment by GregAC** [comment]
Good point @sriyerg will do. Better get round to sorting out a local verible install so I can test it out.

**Comment by GregAC** [comment]
Ok with the OTBN verible lint integrated and passing I'm merging this now as we're keen to get the OTBN RTL into master so we can retire the dev branch ASAP.

I am happy to address any further comments in a follow up PR.

**Pull request merged**